packages/SwingSet/src/kernel/state/kernelKeeper.js

```diff
@@ -1790,41 +1791,15 @@ export default function makeKernelKeeper(
       }
     }
 
-    function compareNumbers(a, b) {
-      return Number(a - b);
-    }
-
-    function compareStrings(a, b) {
-      // natural-sort strings having a shared prefix followed by digits
-      // (e.g., 'ko42' and 'ko100')
-      const [_a, aPrefix, aDigits] = /^(\D+)(\d+)$/.exec(a) || [];
-      if (aPrefix) {
-        const [_b, bPrefix, bDigits] = /^(\D+)(\d+)$/.exec(b) || [];
-        if (bPrefix === aPrefix) {
-          return compareNumbers(aDigits, bDigits);
-        }
-      }
-
-      // otherwise use the default string ordering
-      if (a > b) {
-        return 1;
+    // Perform an element-by-element natural sort.
+    kernelTable.sort((a, b) => {
+      const len = Math.min(a.length, b.length);
```

Comment:
This feels wrong.. actually the original feels wrong too. The `kernelTable` data is mostly created by calling `vatKeeper.dumpState()`, and that returns a 3-tuple of `[kernelSlot, vatID, vatSlot]`. It would make the most sense to use `naturalCompare` separately on each of the three components (krefs, vatIDs, and vrefs all have the same letters-then-digits shape).

I don't know why the original was treating the rows as *six*-tuples. My best guess is it thought `a[0], a[1], a[2]` was individual characters of a kref, so the `ko` prefix would be compared by strings, and the digit suffix would be compared by number. But that doesn't make a lot of sense either.

In any case, I think we want something more like:
```js
kernelTable.sort((a,b) =>
  naturalCompare(a[0], b[0]) ||
  naturalCompare(a[1], b[1]) ||
  naturalCompare(a[2], b[2]) || 0);
```

unless you can make more sense of the old code than me..

Reply:
Yeah, that makes sense. I just realized I typo'd the general form anyway (`a[0]` and `b[0]` rather than `a[i]` and `b[i]`).